sweets/bad.yml

```diff
@@ -0,0 +1,293 @@
+    AllowedValues:
+    - "KMS"
+    - "SSE-S3"
+    Default: "SSE-S3"
+
+  ExpirationInDays:
+    Type: Number
+    Default: 30
+
+  OrgAccessList:
+    Type: CommaDelimitedList
+    Default: ""
+
+  ProjectName:
+    Type: String
+
+  ReplSrcAcct:
+    Type: String
+
+  ReplSrcRole:
+    Type: String
+    Default: ""
+
+  Versioning:
+    Type: String
+    Default: "Enabled"
+    AllowedValues:
+    - "Enabled"
+
+
+Conditions:
+  AllowOrgWideAccess: !Not [!Equals [!Join ["", !Ref OrgAccessList], ""]]
+  EncryptWithKms: !Equals [!Ref Encryption, "KMS"]
+  HasReplSrcRole: !Not [!Equals [!Ref ReplSrcRole, ""]]
+  HasBackfillPrincipal: !Not [!Equals [!Ref BackfillPrincipal, ""]]
+  ExpireNoncurrent: !Not [!Equals [!Ref ExpirationInDays, 0]]
+
+Resources:
+  EncryptionKey:
+    Type: AWS::KMS::Key
+    Condition: EncryptWithKms
+    Properties:
+      Description: !Sub "Encryption Key for ${Account}-${ProjectName} bucket"
+      KeyPolicy:
+        Version: "2012-10-17"
+        Statement:
+            # https://aws.amazon.com/premiumsupport/knowledge-center/update-key-policy-future/
+            # the first statement below disables the key lockout check
+          - Sid: "Allow IAM permissions in archive"
+            Effect: Allow
+            Action:
+              - "kms:*"
+            Principal:
+              AWS:
+                - !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Resource: "*"
+
+            # NOTE Encryption of large files requires decrypt permissions:
+            #      https://aws.amazon.com/premiumsupport/knowledge-center/s3-large-file-encryption-kms-key/
+          - Sid: "Allow encryption by replication role"
+            Effect: Allow
+            Action:
+              - "kms:Encrypt"
+              - "kms:Decrypt"
+              - "kms:GenerateDataKey"
+            Principal:
+              AWS:
+                - !Sub "arn:aws:iam::${ReplSrcAcct}:role/service-role/${ReplSrcRole}"
+                - !If
+                  - HasBackfillPrincipal
+                  - !Ref BackfillPrincipal
+                  - !Ref AWS::NoValue
+            Resource: "*"
+            Condition:
+              StringLike:
+                kms:ViaService: !Sub "s3.${AWS::Region}.amazonaws.com"
+                kms:EncryptionContext:aws:s3:arn: !Sub "arn:aws:s3:::${Account}-${ProjectName}/*"
+
+          - !If
+            - AllowOrgWideAccess
+            - Sid: "Allow decrypt to all in the Organization"
+              Effect: Allow
+              Action:
+                - "kms:Decrypt"
+              Principal: "*"
+              Resource: "*"
+              Condition:
+                StringEquals:
+                  aws:PrincipalOrgID: !Ref OrgAccessList
+            - !Ref AWS::NoValue
+
+  EncryptionKeyAlias:
+    Type: AWS::KMS::Alias
+    Condition: EncryptWithKms
+    Properties:
+      AliasName: !Sub "alias/${ProjectName}/key"
+      TargetKeyId: !Ref EncryptionKey
+
+  Bucket:
+    Type: AWS::S3::Bucket
```

Comment:
*(Lintly/cfn-nag)* W35: S3 Bucket should have access logging configured  <!-- Automatically posted by Lintly -->